### DISCUSSION TITLE
FalkorDB Multitenancy or MultiIndexing

### DISCUSSION THREAD

**Discussion by pathan33** [FalkorDB/FalkorDB#791] [General]
Does FalkorDB really support multitenancy? I’d like to store the nodes and relationships of different clients in separate indexes or assign each client a unique identifier/differentiator within FalkorDB?

**Comment by AviAvni** [comment]
There are 2 options
1. create a graph per client by using different key for each one it can improve perf and be more secure as it is not possible to get by mistake data that is not related to other client
2. Label the nodes with the user identity this is less secure but you can have shared node or relationship between clients

**Reply by pathan33** [reply]
Thanks for your response.

So, to achieve multitenancy, as per this [documentation](https://api.python.langchain.com/en/latest/graphs/langchain_community.graphs.falkordb_graph.FalkorDBGraph.html#langchain_community.graphs.falkordb_graph.FalkorDBGraph.__init__), if I use the different database name, username and password parameters in the __init_falkordb_connection method, then will this help to isolate graphs in FalkorDB.

Please confirm on this!
